python/tvm/driver/tvmc/common.py

```diff
@@ -136,3 +138,46 @@ def tracker_host_port_from_cli(rpc_tracker_str):
         logger.info("RPC tracker port: %s", rpc_port)
 
     return rpc_hostname, rpc_port
+
+
+def parse_shape_string(inputs):
+    """Parse an input shape dictionary string to a usable dictionary.
+
+    Parameters
+    ----------
+    inputs: str
+        A string of the form "name:num1xnum2x...xnumN,name2:num1xnum2xnum3" that indicates
+        the desired shape for specific model inputs.
+
+    Returns
+    -------
+    shape_dict: dict
+        A dictionary mapping input names to their shape for use in relay frontend converters.
+    """
+    inputs = inputs.replace(" ", "")
+    # Check if the passed input is in the proper format.
+    valid_pattern = re.compile("(\w+:(\d+(x|X))*(\d)+)(,(\w+:(\d+(x|X))*(\d)+))*")
+    result = re.fullmatch(valid_pattern, inputs)
```

Comment:
- Seems doesn't match negative numbers such as `data:-1x3x224x224`.
- Same comment as the one I left in another PR: This syntax is not straightforward to many users. I would suggest using either JSON or Python syntax.

Reply:
@comaniac I agree the syntax is a little confusing, but discussed it with Jocelyn and there are some tradeoffs that would be made using a more Python/JSON like syntax. Since we're passing the shapes argument via command line, we cant use spaces or semicolons unless the whole argument is wrapped in quotes. For example, if we used a python like syntax such as `tvmc compile --shapes=data:-1,3,224,224 data2:10,10` the space would prevent argparse from recognizing it as a single argument. We could require quote wrapping but I personally think that is more likely to cause issues than using `x` instead of `,` to separate dimensions. Do you have a recommendation for the input format that you prefer or are you ok with always wrapping in quotes?

Reply:
To be honest, I'm totally fine with quotes in the command line, and we actually use it in TVMC already. For example, when specifying target, you cannot avoid using quotes for targets like `llvm -mcpu=core-axv2`. In fact, although we haven't fully landed the new target, the ultimate goal is to get rid of plain string target format and support JSON format. IMHO, it would be better if the format/syntax are consistent.

Meanwhile, if having quotes in the command line is a concern, it might be worthwhile to consider taking a command line file. See https://docs.python.org/3/library/argparse.html#fromfile-prefix-chars for reference. In this case, we can have:

```
# shapes.txt
--input-shapes
data:[-1,3,224,224] data2: [10,10]
```

```
tvmc compile @shapes.txt
```

Reply:
> See https://docs.python.org/3/library/argparse.html#fromfile-prefix-chars for reference

That's really cool @comaniac I didn't know this.

I agree that `data:[-1,3,224,224] data2: [10,10]` syntax looks clearer, IMHO.